src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrType.cs

```diff
@@ -188,9 +189,14 @@ private static System.Type GetProxyType(System.Type type)
             return assembly.GetType(proxyName);
         }
 
-        private static ReadOnlyCollection<DkmClrEvalAttribute> GetEvalAttributes(Type type)
+        private static ReadOnlyCollection<DkmClrEvalAttribute> GetEvalAttributes(DkmClrAppDomain appDomain, Type type)
         {
             var reflectionType = ((TypeImpl)type).Type;
+            if (appDomain.TypeToEvalAttributesMap.TryGetValue(reflectionType, out var cachedAttributes))
```

Comment:
Actually executing `GetEvalAttributes` is allocation wise very expensive. The DKM layer creates `DkmClrType` instances for the same `Type` many many times. That means we're doing so much unnecessary work. This simple cache saves 200MB allocations in the case the unit test OOMs. Not sure how much it saves in the case where it actually completes